providers/base/bin/prime_offload_tester.py

```diff
@@ -0,0 +1,334 @@
+        """
+        Use to check provided command is executed on specific GPU.
+
+        .. note::
+            While setting prime offload environment such as DRI_PRIME,
+            the process will be listed under kernel debug interface.
+            The location of kernel debug interface is
+            /sys/kernel/debug/dri/<card id>,
+            and the process could be found in
+            /sys/kernel/debug/dri/<card id>/clients
+
+        :param cmd: command that running under prime offload
+        :type cmd: str
+
+        :param card_id: card id of dri device
+        :type card_id: str
+
+        :param card_name: card name of dri device
+        :type card_name: str
+        """
+        cmd_without_args = cmd.split(' ')[0]
+        for index in range(1, 11):
+            time.sleep(2)
+            try:
+                read_clients_cmd = \
+                    "sudo cat /sys/kernel/debug/dri/{}/clients".format(card_id)
+                clients = subprocess.run(read_clients_cmd, shell=True,
+                                         stdout=subprocess.PIPE,
+                                         universal_newlines=True)
+                if cmd_without_args in clients.stdout:
+                    self.logger.info("Checking success:")
+                    self.logger.info("  Offload process:[{}]".format(cmd))
+                    self.logger.info("  Card ID:[{}]".format(card_id))
+                    self.logger.info("  Device Name:[{}]".format(card_name))
+                    self.check_result = PrimeOffloaderError.NO_ERROR
+                    return
+                self.check_result = PrimeOffloaderError.OFFLOAD_FAIL
+            except (OSError, TypeError):
+                # Missing file or permissions?
+                self.logger.info("Couldn't open file for"
+                                 "reading clients of dri device")
+                self.check_result = PrimeOffloaderError.OFFLOAD_FAIL
+                return
+        self.logger.info("Checking fail:")
+        self.logger.info("  Couldn't find process [{}]".format(cmd) +
+                         " running after check {} times".format(index))
+        self.check_result = PrimeOffloaderError.OFFLOAD_FAIL
+
+    def check_nv_offload_env(self):
+        """
+        prime offload of nvidia driver is limited.
+        Only on-demand mode is supported.
+        """
+        # nvidia-smi ship with NVIDIA GPU display drivers on Linux
+        # https://developer.nvidia.com/nvidia-system-management-interface
+        # check prime-select to make sure system with nv driver.
+        # If there is no nv driver, prime offload is fine for other drivers.
+        ps = subprocess.run("whereis prime-select | cut -d ':' -f 2",
+                            shell=True,
+                            stdout=subprocess.PIPE,
+                            universal_newlines=True)
+        if 'prime-select' not in ps.stdout:
+            return PrimeOffloaderError.NO_ERROR
```

Comment:
Are you sure that not having `prime-select` available in the system means "NO ERROR"? Feels like the opposite.

Reply:
For prime offloading, it only works on non Nvidia mode. Therefore, without `prime-select` (means no Nvidia driver installed).